extensions/amp-ooyala-player/0.1/amp-ooyala-player.js

```diff
@@ -0,0 +1,139 @@
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {user} from '../../../src/log';
+import {removeElement} from '../../../src/dom';
+
+class AmpOoyalaPlayer extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://player.ooyala.com', opt_onLayout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+
+    const embedCode = user().assert(
+        this.element.getAttribute('data-embedcode'),
+        'The data-embedcode attribute is required for <amp-ooyala-player> %s',
+        this.element);
+    const pCode = user().assert(
+        this.element.getAttribute('data-pcode'),
+        'The data-pcode attribute is required for <amp-ooyala-player> %s',
+        this.element);
+    const playerId = user().assert(
+        this.element.getAttribute('data-playerid'),
+        'The data-playerid attribute is required for <amp-ooyala-player> %s',
+        this.element);
+
+    let src = 'https://player.ooyala.com/iframe.html?platform=html5-priority';
+    const playerVersion = this.element.getAttribute('data-playerversion') || '';
+    if (playerVersion.toLowerCase() == 'v4') {
+      src = 'https://player.ooyala.com/static/v4/stable/latest/skin-plugin/amp_iframe.html?pcode='
+        + encodeURIComponent(pCode);
+      const configUrl = this.element.getAttribute('data-config');
+      if (configUrl) {
+        src += '&options[skin.config]=' + encodeURIComponent(configUrl);
+      }
+    }
+
+    src += '&ec=' + encodeURIComponent(embedCode) +
+      '&pbid=' + encodeURIComponent(playerId);
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.src = src;
+    this.applyFillContent(iframe);
+
+    this.element.appendChild(iframe);
+    return this.loadPromise(iframe);
+  }
+
+  /** @override */
+  play(unusedIsAutoplay) {
+    this.iframe_.contentWindow.postMessage('play', '*');
+  }
+
+  /** @override */
+  pause() {
+    this.iframe_.contentWindow.postMessage('pause', '*');
+  }
+
+  /** @override */
+  mute() {
+    this.iframe_.contentWindow.postMessage('mute', '*');
+  }
+
+  /** @override */
+  unmute() {
+    this.iframe_.contentWindow.postMessage('unmute', '*');
+  }
+
+  /** @override */
+  pauseCallback() {
+    if (this.iframe_) {
+      this.pause();
+    }
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    return true;
+  }
+
+  /** @override */
+  unlayoutOnPause() {
+    return false;
+  }
+
+  /** @override */
+  supportsPlatform() {
+    return true;
+  }
+
+  /** @override */
+  isInViewport() {
```

Comment:
Are you sure? VideoInterface has isInViewPort and it doesn't have isInteractive.  Actually, there is no isInteractive strings at all in the source code.